config/puma.rb

```diff
@@ -1,3 +1,26 @@
 # frozen_string_literal: true
 
 plugin :statsd
+
+# Configure "min" to be the minimum number of threads to use to answer
+# requests and "max" the maximum.
+#
+# The default is "0, 16".
+#
+threads 32, 32
+
+# === Cluster mode ===
+
+# How many worker processes to run.
+#
+# The default is "0".
+#
+workers 2
+
+# Verifies that all workers have checked in to the master process within
+# the given timeout. If not the worker process will be restarted. Default
+# value is 60 seconds.
+#
+worker_timeout 60
+
+preload_app!
```

Comment:
Normally when you preload the app, you also set up the connections to the database. 

```
on_worker_boot do
  ActiveRecord::Base.establish_connection # can also connect Resque/redis here since we use a lot of that
end
```

This usually helps reduce memory usage (https://devcenter.heroku.com/articles/deploying-rails-applications-with-the-puma-web-server#on-worker-boot)

Reply:
you also need to disconnect any network connections before worker boot so they don't leak after fork().  I would say don't preload in this PR, do it separately.